### PR TITLE
[ADD] add the module sale_automatic_workflow_cancel this module give the...

### PR DESCRIPTION
**Pull request by sebastienbeau**
... posibility to cancel automatically unconfirm order after some delay and depending of the payment method

### CODE REVIEW

File: sale_automatic_workflow_cancel/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+###############################################################################
+
+{'name': 'Sale Automatic Workflow Cancel',
+ 'version': '0.0.1',
+ 'author': 'Akretion',
+ 'website': 'www.akretion.com',
+ 'license': 'AGPL-3',
+ 'category': 'Generic Modules',
+ 'description': """Depending of the configuration of the payment
```
Comment (pedrobaeza) [inline sale_automatic_workflow_cancel/__openerp__.py]: Depending s/of/on

File: sale_automatic_workflow_cancel/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ method sale order will be automatically cancelled after some delay.
+ For example you can configure that every sale order paid buy check will
```
Comment (pedrobaeza) [inline sale_automatic_workflow_cancel/__openerp__.py]: s/buy/by

File: sale_automatic_workflow_cancel/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ be cancelled after 90 days.
```
Comment (pedrobaeza) [inline sale_automatic_workflow_cancel/__openerp__.py]: You should complete it with: `...if no payment is done`

File: sale_automatic_workflow_cancel/payment.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        ids = self.search(cr, uid, domain, context=context)
+        for method in self.browse(cr, uid, ids, context=context):
+            sale_domain = self._get_sale_domain(
+                cr, uid, method, context=context)
+            order_ids = order_obj.search(
+                cr, uid, sale_domain, context=context)
+            _logger.debug('Cancel %s Sale Order for method %s'
+                          % (len(order_ids), method.name))
+            for order_id in order_ids:
+                with commit(cr):
```
Comment (pedrobaeza) [inline sale_automatic_workflow_cancel/payment.py:73]: Why do you need commit? Better to embrace in a `try:` for each order so that one errored order doesn't affect the others

Comment (guewen) [inline sale_automatic_workflow_cancel/payment.py:73]: I think it's fine to commit in this context:
- we are in a cron
- it will lock only 1 sale order at a time and for a short duration, not locking a long range and blocking users
- an ordered error is rollbacked and the process continues with the next
- if the cron is interrupted, all the already cancelled orders will already be committed.

File: sale_automatic_workflow_cancel/payment.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+class PaymentMethod(orm.Model):
+    _inherit = 'payment.method'
+
+    _columns = {
+        'days_before_cancel': fields.integer(
+            'Days before cancel',
+            help="After 'n' days, if the 'Import Rule' is not fulfilled, the "
+                 "the sale order (or the import of the sale order) will "
+                 "be canceled."),
```
Comment (guewen) [inline sale_automatic_workflow_cancel/payment.py:40]: The help message seems unrelated to the actual implementation.

File: sale_automatic_workflow_cancel/payment.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        order_obj = self.pool['sale.order']
+        domain = self._get_method_domain(cr, uid, context=context)
+        ids = self.search(cr, uid, domain, context=context)
+        for method in self.browse(cr, uid, ids, context=context):
+            sale_domain = self._get_sale_domain(
+                cr, uid, method, context=context)
+            order_ids = order_obj.search(
+                cr, uid, sale_domain, context=context)
+            _logger.debug('Cancel %s Sale Order for method %s'
+                          % (len(order_ids), method.name))
```
Comment (guewen) [inline sale_automatic_workflow_cancel/payment.py:71]: Do not use string substitution in loggers.
The correct form is:

```
_logger.debug('Cancel %s Sale Order for method %s', len(order_ids), method.name)
```

File: sale_automatic_workflow_cancel/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+{'name': 'Sale Automatic Workflow Cancel',
+ 'version': '0.0.1',
+ 'author': 'Akretion',
```
Comment (gurneyalex) [inline sale_automatic_workflow_cancel/__openerp__.py:25]: can you add "Odoo Community Association (OCA)" ?

**Comment by pedrobaeza** [comment]
@sebastienbeau, are you going to continue this work?

**Comment by pedrobaeza** [comment]
As there's no answer, I close. Feel free to reopen in case there's interest.
